lib/src/widgets/animated_button.dart

```diff
@@ -130,6 +126,40 @@ class _AnimatedButtonState extends State<AnimatedButton>
     }
   }
 
+  // initializes width and size animation
+  void _initWidth(BuildContext context) {
+    final theme = Theme.of(context);
+    final fontSize = theme.textTheme.button.fontSize;
+
+    var renderParagraph = RenderParagraph(
+      TextSpan(
+        text: widget.text,
+        style: TextStyle(
+          fontSize: fontSize,
+          fontWeight: theme.textTheme.button.fontWeight,
+          letterSpacing: theme.textTheme.button.letterSpacing,
+        ),
+      ),
+      textDirection: TextDirection.ltr,
+      maxLines: 1,
+    );
+
+    renderParagraph.layout(BoxConstraints(minWidth: 120.0));
+
+    // text width based on fontSize, plus 24.0 for padding
+    var textWidth = renderParagraph.getMinIntrinsicWidth(fontSize).ceilToDouble() + 24.0;
+
+    // button width is min 120.0 and max 240.0
+    _width = textWidth > 120.0 && textWidth < 240.0 ? textWidth :
+      textWidth >= 240.0 ? 240.0 : 120.0;
```

Comment:
Please use dart formatter next time you add something long (`Ctrl` + `Alt` + `F` shortcut in vscode)